packages/cli/src/ui/components/SettingsDialog.tsx

```diff
@@ -250,65 +224,44 @@ export function SettingsDialog({
               selectedScope,
             );
 
+            // Force re-render by updating previewSettings state
+            // This is necessary because the settings prop was mutated but React doesn't know
+            setPendingSettings((prev) =>
+              setPendingSettingValueAny(key, newValue, prev),
+            );
+
             // Special handling for vim mode to sync with VimModeContext
             if (key === 'general.vimMode' && newValue !== vimEnabled) {
-              // Call toggleVimEnabled to sync the VimModeContext local state
               toggleVimEnabled().catch((error) => {
-                console.error('Failed to toggle vim mode:', error);
+                debugLogger.log(
+                  '[DEBUG SettingsDialog] Failed to toggle vim mode:',
```

Comment:
don't change this to a debugLogger.log. Need to determine whether it is a user visible error or not.